src/Sylius/Bundle/ShopBundle/spec/EventListener/NonChannelLocaleListenerSpec.php

```diff
@@ -176,10 +185,15 @@ function it_throws_not_found_exception_if_request_locale_is_not_present_in_provi
         $request->attributes = new ParameterBag();
 
         $localeProvider->getAvailableLocalesCodes()->willReturn(['ga', 'ga_IE']);
+        $localeProvider->getDefaultLocaleCode()->willReturn('ga');
 
-        $this
-            ->shouldThrow(NotFoundHttpException::class)
-            ->during('restrictRequestLocale', [$event])
-        ;
+        $router->generate(
+            'sylius_shop_homepage',
+            ['_locale' => 'ga']
+        )->willReturn('/ga/');
+        $redirect = new RedirectResponse('/ga/');
+
+        $this->restrictRequestLocale($event);
+        $event->setResponse($redirect)->shouldHaveBeenCalledOnce();
```

Comment:
Shouldn't it be placed before a `restrictRequestLocale` call?

Reply:
After making the change it didn't work.
I did some research and found the documentation that talks about the subject.
And so I think my first version was right.

https://www.phpspec.net/en/latest/manual/prophet-objects.html